charger/fritzdect.go

```diff
@@ -48,28 +49,51 @@ func NewFritzDECT(uri, ain, user, password string, standbypower float64) (*Fritz
 	fd := &FritzDECT{
 		fritzdect:    fritzdect,
 		standbypower: standbypower,
+		staticmode:   standbypower < 0,
 	}
 	return fd, nil
 }
 
 // Status implements the api.Charger interface
 func (c *FritzDECT) Status() (api.ChargeStatus, error) {
 	// present 0/1 - DECT Switch connected to fritzbox (no/yes)
-	var present int64
+	var present bool
+	var Ison bool
+	var power float64
+
 	resp, err := c.fritzdect.ExecCmd("getswitchpresent")
 	if err == nil {
-		present, err = strconv.ParseInt(resp, 10, 64)
+		present, err = strconv.ParseBool(resp)
 		if err != nil {
 			return api.StatusNone, err
 		}
 	}
+	if !present {
+		return api.StatusNone, api.ErrNotAvailable
+	}
 
-	power, err := c.fritzdect.CurrentPower()
+	if c.staticmode {
+		Ison, err = c.Enabled()
+		if err != nil {
+			return api.StatusNone, err
+		}
+	} else {
+		power, err = c.fritzdect.CurrentPower()
+		if err != nil {
+			return api.StatusNone, err
+		}
+	}
 
 	switch {
```

Comment:
ich fänds besser die beiden modes komplett zu trennen als jedes statement durcheinander zu würfeln. Nach ison kannst Du ja schon direkt `B` vs `C` zurück geben. Wie wäre es not present als `A` zu handeln?